packages/legend-application-studio/src/stores/editor/EditorStore.ts

```diff
@@ -337,6 +350,24 @@ export class EditorStore implements CommandRegistrar {
     this.showSearchElementCommand = val;
   }
 
+  getSourceInfo(): ProjectQuerySDLC | undefined {
+    if (this.mode === EDITOR_MODE.VIEWER) {
+      return {
+        projectId: this.sdlcState.activeProject.projectId,
```

Comment:
should we maybe put this in the EditorMode class ? 
This would break in showcaseeditorMode as well as when viewer mode is using gav coordinates right???

Reply:
Lets be careful using using methods that throw here. most likely we should safely call this method and return undefined if we can't calculate source